File: learning-javascript/leet-code/src/remove-duplicates-from-sorted-array/remove-duplicates-from-sorted-array.ts
/**
 * time complexity: O(n)
 * @param nums 
 * @returns 
 */

export function removeDuplicates(nums: number[]): number {
	let i = 1;
	let prevValue = nums[0];
	while(i < nums.length) {
		if(nums[i] === prevValue) {
			nums.splice(i, 1);
			continue;
		}
		prevValue = nums[i];
		i++;
	}
	return nums.length;
};

export function removeDuplicates2(nums: number[]): number {
	let i = 1;
	let prevValue = nums[0];
    let dupe = {
        value: null,
        count: 0,
    };
	while(i < nums.length) {
		if(nums[i] === prevValue) {
			if(prevValue === dupe.value){
					dupe.count++;
					if(dupe.count > 2) {
				nums.splice(i, 1);
				continue;
					}
			} else {
					dupe.count = 2;
					dupe.value = prevValue;
			}
		}
		prevValue = nums[i];
		i++;
	}
	return nums.length;
};

File: learning-python/leet-code/src/remove_duplicates_from_sorted_array/remove_duplicates_from_sorted_array.py
def removeDuplicates(nums) :
	i = 1
	prevValue = nums[0]
	while i < len(nums):
		if nums[i] == prevValue:
			nums.remove(prevValue)
			continue
		prevValue = nums[i]
		i += 1
	return len(nums)

def removeDuplicates2(nums):
	i = 1
	prevValue = nums[0]
	dupe = {
			'value': '',
			'count': 0,
	}
	while i < len(nums):
		if nums[i] == prevValue:
				if prevValue == dupe['value']:
						dupe['count']+= 1
						if dupe['count'] > 2:
								nums.remove(prevValue)
								continue
				else:
						dupe['count'] = 2
						dupe['value'] = prevValue
		prevValue = nums[i]
		i+=1
	return len(nums)